kinto/core/initialization.py

```diff
@@ -252,6 +253,23 @@ def setup_cache(config):
     config.registry.heartbeats['cache'] = heartbeat
 
 
+def setup_hawk_auth(config):
+    settings = config.get_settings()
+    
+    if 'hawk.enabled' in settings and settings['hawk.enabled']:
+        def on_new_request(event):
+            request = event.request
+            authenticator = HawkAuth(settings['hawk.id'],   
+                                     settings['hawk.secret'], 
+                                     settings['hawk.algo'],
+                                     config)
+            if not authenticator.authenticate(request):
```

Comment:
We should plug it to multiauth so that we can use multiple authentication mecanisms in parallel.